.github/workflows/ci.yaml

```diff
@@ -182,3 +182,50 @@ jobs:
         run-e2e: true
         ngrok-api-key: ${{ secrets.NGROK_CI_API_KEY }}
         ngrok-authtoken: ${{ secrets.NGROK_CI_AUTHTOKEN }}
+
+  e2e-multi-namespace:
+    runs-on: ubuntu-latest
+    needs:
+    - changes
+    - kubebuilder-diff
+    - lint
+    - e2e
+    concurrency:
+      group: e2e
+      cancel-in-progress: false
+    timeout-minutes: 30
+    if: |
+      (github.repository == 'ngrok/ngrok-operator') &&
+      (
+        (github.event_name == 'push' && github.ref_name == 'main') ||
+        (github.event_name == 'merge_group')
+      ) &&
+      (
+        (needs.changes.outputs.go == 'true') ||
+        (needs.changes.outputs.charts == 'true') ||
+        (needs.changes.outputs.chartyaml == 'true') ||
+        (needs.changes.outputs.tests == 'true') ||
+        (needs.changes.outputs.make == 'true')
+      )
+    steps:
+    - uses: actions/checkout@v6
+    - uses: nixbuild/nix-quick-install-action@v34
+    - uses: DeterminateSystems/magic-nix-cache-action@b8276522d77f21bf19d3574e5bc99186a6c7aa6c
+    - name: Create Kind Cluster
+      run: nix develop --command make kind-create
+    - name: Deploy
+      run: nix develop --command make deploy_multi_namespace
+      env:
+        NGROK_API_KEY: ${{ secrets.NGROK_CI_API_KEY }}
+        NGROK_AUTHTOKEN: ${{ secrets.NGROK_CI_AUTHTOKEN }}
+    - name: Test
+      run: nix develop --command make e2e-tests-multi-ns
+    - name: Cleanup
+      if: always()
+      run: |
+        echo "Deleting kubernetesoperators..."
+        kubectl -n namespace-a delete kubernetesoperator ngrok-operator-a || true
+        kubectl -n namespace-b delete kubernetesoperator ngrok-operator-b || true
+        echo "Uninstalling the operators..."
+        helm uninstall ngrok-operator-a -n namespace-a || true
+        helm uninstall ngrok-operator-b -n namespace-b || true
```

Comment:
would it be worth removing the `|| true` so it actually fails if they error? Or if we don't care about ensuring these uninstalls work and the goal is to just reset kind we can just delete the kind cluster